staging/src/kubevirt.io/client-go/api/v1/hyperconvergedconfig.go

```diff
@@ -0,0 +1,39 @@
+package v1
+
+// This code is copied from
+// https://github.com/kubevirt/hyperconverged-cluster-operator/blob/master/pkg/apis/hco/v1beta1/hyperconverged_types.go
+// in order to avoid dependency loops
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// HyperConvergedConfig is the schema for pod configuration that is going to
+// be propagated by HCO to its managed operators.
+//
+// +k8s:openapi-gen=true
+type HyperConvergedConfig struct {
+	// nodeSelector is the node selector applied to the relevant kind of pods
+	// It specifies a map of key-value pairs: for the pod to be eligible to run on a node,
+	// the node must have each of the indicated key-value pairs as labels
+	// (it can have additional labels as well).
+	// See https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector
+	// +kubebuilder:validation:Optional
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// affinity enables pod affinity/anti-affinity placement expanding the types of constraints
+	// that can be expressed with nodeSelector.
+	// affinity is going to be applied to the relevant kind of pods in parallel with nodeSelector
+	// See https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity
+	// +kubebuilder:validation:Optional
+	// +optional
+	Affinity *corev1.Affinity `json:"affinity,omitempty"`
+
+	// tolerations is a list of tolerations applied to the relevant kind of pods
+	// See https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/ for more info.
+	// These are additional tolerations other than default ones.
+	// +kubebuilder:validation:Optional
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
```

Comment:
Would you be so kind and use [marker](https://kubernetes.io/docs/reference/using-api/api-concepts/#merge-strategy) on this field. It will sillent kube-openapi.

Reply:
Odd that k8s doesn't apply a marker to this field either?

@tiraboschi All HCO-based projects should do this the same way. Please advise.

Reply:
`// +listType=set` ?

Reply:
:rofl: I'll accept this as an authoritative answer despite the question mark...

Reply:
I'm trying to sync CDI with the code in HCO.
I get slightly more errors with listType=set
````
Type: "FieldValueInvalid",
Message: "Invalid value: \"object\": must be a scalar or atomic type as item of a list with x-kubernetes-list-type=set",
Field: "spec.versions[1].schema.openAPIV3Schema.properties[spec].properties[workload].properties[tolerations].items.type",
````

Is there something about the HCO code that handles this issue?

Reply:
I think it should be ` //+listType=map`